security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -630,12 +637,29 @@ func (sc *SecretCache) rotate(updateRootFlag bool) {
 		if sc.shouldRotate(&secret) {
 			atomic.AddUint64(&sc.secretChangedCount, 1)
 			// Send the notification to close the stream if token is expired, so that client could re-connect with a new token.
-			if sc.isTokenExpired(&secret) {
-				cacheLog.Debugf("%s token expired", logPrefix)
+			isTokenExpired := sc.isTokenExpired(&secret)
+			if isTokenExpired {
+				cacheLog.Infof("%s token expired", logPrefix)
+
 				// TODO(myidpt): Optimization needed. When using local JWT, server should directly push the new secret instead of
-				// requiring the client to send another SDS request.
-				sc.callbackWithTimeout(connKey, nil /*nil indicates close the streaming connection to proxy*/)
-				return true
+
+				switch sc.fetcher.CaClient.(type) {
```

Comment:
Please do not do this, this just make the interface and the abstract of the CA client useless.

Reply:
ok i will directly use the interface function to do the reset